src/Bicep.LangServer/Telemetry/TelemetryHelper.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Configuration;
+using Bicep.LanguageServer.Providers;
+
+namespace Bicep.LanguageServer.Telemetry
+{
+    public class TelemetryHelper
+    {
+        public const string LinterEnabledSetting = "core.enabled";
+
+        public static void SendTelemetryOnBicepConfigChange(RootConfiguration prevConfiguration, RootConfiguration curConfiguration, ILinterRulesProvider linterRulesProvider, ITelemetryProvider telemetryProvider)
+        {
+            foreach (var telemetryEvent in GetTelemetryEventsForBicepConfigChange(prevConfiguration, curConfiguration, linterRulesProvider))
+            {
+                telemetryProvider.PostEvent(telemetryEvent);
+            }
+        }
+
+        public static IEnumerable<BicepTelemetryEvent> GetTelemetryEventsForBicepConfigChange(RootConfiguration prevConfiguration, RootConfiguration curConfiguration, ILinterRulesProvider linterRulesProvider)
+        {
+            bool prevLinterEnabledSettingValue = prevConfiguration.Analyzers.GetValue(LinterEnabledSetting, true);
+            bool curLinterEnabledSettingValue = curConfiguration.Analyzers.GetValue(LinterEnabledSetting, true);
+
+            if (!prevLinterEnabledSettingValue && !curLinterEnabledSettingValue)
+            {
+                return Enumerable.Empty<BicepTelemetryEvent>();
+            }
+
+            List<BicepTelemetryEvent> telemetryEvents = new();
+
+            if (prevLinterEnabledSettingValue != curLinterEnabledSettingValue)
```

Comment:
I don't understand the reason for this condition.  If enabled has not changed, don't we want to send info on which configs changed?